### ISSUE TITLE
Platform Feature Matrix

### ISSUE THREAD

**Issue by lrusak** [xbmc/xbmc#14967]
## Platform Feature Matrix

Here we can keep track of what platforms have what features available

This feature matrix is quite a complex beast and is difficult to represent everything exactly.

`x` indicates that the platform does support the feature
`-` indicates that the feature depends on the platform configuration (see extra options to the right of the platforms)

|                  | iOS | OSX | Windows | UWP | Android | Amlogic | GBM | RPi | Wayland | X11 |   | GLES | GL | ALSA | Pulse | VAAPI | VDPAU |
|------------------|:---:|:---:|:-------:|:---:|:-------:|:-------:|:---:|:---:|:-------:|:---:|---|:----:|:--:|:----:|:-----:|:-----:|:-----:|
| **HDR**              |     |     |         |     |         |         |     |     |         |     |   |      |    |      |       |       |       |
| **Tone Mapping**     | [1] | [1] |    x    |  x  |   [1]   |   [1]   |  -  | [1] |    -    |  -  |   |   x  |  x |      |       |       |       |
| **10 bit**           |     |  x  |    x    |  x  |         |         |  -  |     |    -    |  -  |   |      |  x |      |       |       |       |
| **HEVC**             |     |     |    x    |  x  |    x    |    x    |  -  |     |    -    |  -  |   |      |    |      |       |   x   |   x   |
| **VP9**              |     |     |         |     |    x    |    x    |  -  |     |    -    |  -  |   |      |    |      |       |   x   |       |
| **Color Management** |     |     |         |     |         |         |  -  |     |    -    |  -  |   |      |  x |      |       |       |       |
| **HBR**              |     |     |    x    |  x  |    x    |    -    |  -  |  -  |    -    |  -  |   |      |    |   x  |       |       |       |
| **DPMS**             |     |  x  |    x    |  x  |         |         |  x  |     |         |  x  |   |      |    |      |       |       |

 1) Tone mapping may not  be supported by hardware decoding/rendering but is available via software   decoding/rendering

Table was created using https://www.tablesgenerator.com/markdown_tables

------------------------------

Team Members - Please help me keep this up to date and fill in anything that I missed and add other features as you see fit.

**Comment by pkerling** [comment]
Wouldn't this be more fit for the wiki?

**Comment by MartijnKaijser** [comment]
Or generated from Doxygen / Github markup. 👍 for the idea

**Comment by lrusak** [comment]
My main thinking for putting it here is so that it can be referenced in a development environment for the purpose of development. People can look here to see that something isn't supported on a platform and can then dig into why it is so. We can also keep track of when features are added for specific platforms. 💡 

I stay away from the wiki so if someone would rather do it there then give'r 👍

**Comment by MartijnKaijser** [comment]
Then it should be put as an .md file under docs and not as issue ticket cause i mentioned that will get burred.

**Comment by lrusak** [comment]
> Then it should be put as an .md file under docs and not as issue ticket cause as i mentioned that will get burred.

Then a PR would have to be created for every change. Using a github issue will allow it to be updated on the fly.

**Comment by pkerling** [comment]
Hmm I also think it's a great idea to do this (sorry if I sounded blunt, wasn't intended), but GH issues just isn't the place. I think it's practically equivalent to the wiki except that it will get buried even more quickly under all the bug reports.

I'd say either do a file under docs and accept that you need to PR or just use the wiki. Maybe there's another superior way?

**Comment by lrusak** [comment]
Then let's just use this as a proving ground to discuss the best way to display it. Markdown is rather limited for what you can do with a table. mediawiki tables are better but I'm not even convinced that a table is the best way to represent it.

The problem is that there is so many cases where things need exceptions. Like for example, on GBM you can take screenshots but if you are playing a video with the DRMPRIME decoder then you won't capture the video frame, etc.

We almost need a feature matrix per platform but that would get very redundant to update with new features.

Ideas?

**Comment by pkerling** [comment]
I know it's not optimal, but the Wikipedia comparison pages seem OK

https://en.wikipedia.org/wiki/Comparison_of_wiki_software#Features_1

**Comment by FernetMenta** [comment]
I think most people get confused by this matrix. You need in depth knowledge to interpret the "-" sign. The average person would interpret the matrix in a way that X11 or GBM don't support much because they have no idea about GL,GLES,VAAPI, etc.

Why do you think OSX does not support hevc? It is supported by sw decoding and vtb.

**Comment by lrusak** [comment]
> I think most people get confused by this matrix. You need in depth knowledge to interpret the "-" sign. The average person would interpret the matrix in a way that X11 or GBM don't support much because they have no idea about GL,GLES,VAAPI, etc.

I agree, that's why I tried to make a legend explaining what the options are :smile_cat: however, I'm still not sure what the best way to format it is.

> Why do you think OSX does not support hevc? It is supported by sw decoding and vtb.

I left it blank because I don't really know what OSX or iOS supports. I made a comment that team members can edit it :wink:

**Comment by wsnipex** [comment]
how about using x[with GL] instead of -?

**Comment by kib** [comment]
We used something like that in the past on the wiki as well: https://kodi.wiki/view/Skin_features_compatibility_list

**Comment by lrusak** [comment]
Ok I started a media wiki page using dynamic page layouts. It's kinda clunky IMO but it works I guess.

If any wiki people want to take a look -> https://kodi.wiki/view/Platform_Feature_Matrix

**Comment by afedchin** [comment]
@lrusak windows and UWP support VP9 hw decoding.

**Comment by malard** [comment]
does this need a refresh or is it up to date?
